### PR TITLE
Handling breaking changes in ZwiftHacks SVG plus extra attributes in rider status

### PR DESCRIPTION
**Pull request by jeroni7100**
Hi Andy, 

Don't auto merge -- you only want to consider the changes to server/map.js, server/pooledRider.js, server/rider.js

I made some breaking changes to the svg map from zwifthacks.com which map.js now handles.

In pooledRider.js and rider.js I have added extra rider status information for the tracked riders. ZwiftMap uses this information.

### CODE REVIEW

File: server/rider.js [hunk 1]
```diff
@@ -178,11 +187,46 @@ class Rider extends EventEmitter {
...
+            this.static = {
+                weight: profile.weight
+            }
+        })
+        .catch(ex => {
+            console.log(`Failed to get profile for ${this.riderId}${errorMessage(ex)}`);
+        });
+  }
+
+  addExtra(status) {
```
Comment (Ogadai) [inline server/rider.js]: Does `addExtra()` do something that I'm not seeing? It looks like it returns a copy of the same object.

File: server/rider.js [hunk 2]
```diff
@@ -178,11 +187,46 @@ class Rider extends EventEmitter {
     return positions.concat(ghostPositions);
   }
 
+  getStatic() {
```
Comment (Ogadai) [inline server/rider.js]: Is `getStatic()` needed? It looks like the one from pooledRider.js

**Comment by jeroni7100** [comment]
Let me check before you do anything. I had to rebase my fork and may have messed up somehow even if I checked carefully. I'll let you know.

Regards,
Jesper

________________________________

**Comment by Ogadai** [comment]
Hi Jesper,

I think I've pulled in the changes you need here:
https://github.com/Ogadai/zwift-second-screen/commit/b7165224adcd7732f4720feb9236036e976a20e7

There's the map fix, and the extra rider status fields.

I haven't been able to test it's working with a real rider yet, so please verify and let me know,

Cheers,
Andy

**Comment by jeroni7100** [comment]
Hi Andy,

It looks fine. I have tested the rider status code together with ZwiftMap.

Can you add the following to the return value in riderPromise in rider.js:

cadence: status.cadence,


Ideally, I would like to have all of the status attributes supported by
zwift-mobile-api returned in riderPromise rather than them being
cherrypicked. I had this in an earlier version of my fork but must admit
that I have screwed up the version control so I will have scrap that fork
and try again to make a proper pull request.

Regards,
Jesper

**Pull request closed without merging**